### PR TITLE
Radha

### PR DESCRIPTION
**Pull request by RadhaArul**
I have fixed the Find and created Global variable class for garage capacity and Hourly Rate

### CODE REVIEW

**Review by cisci86 — CHANGES_REQUESTED** [review]
Make change as disscused

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs [hunk 1]
```diff
@@ -12,12 +12,15 @@ namespace Garage_2._0.Controllers.VehiclesController
...
-
         Vehicle[] parkingSpots;
-
-        public VehiclesController(GarageVehicleContext context)
+        IConfiguration _config;
+        public VehiclesController(GarageVehicleContext context,IConfiguration config)
         {
             _context = context;
+            _config = config;
+            Global.Garagecapacity = _config.GetValue<int>("GarageCapacity:Capacity");
```
Comment (cisci86) [inline Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: Is it possible to have the code after the "=" in the Global class instead?

**Review by cisci86 — COMMENTED** [review]
See comment in the code

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs [hunk 2]
```diff
@@ -12,14 +12,15 @@ namespace Garage_2._0.Controllers.VehiclesController
     public class VehiclesController : Controller
     {
         private readonly GarageVehicleContext _context;
-
         Vehicle[] parkingSpots;
-
-        IConfiguration _iConfig;
-        public VehiclesController(GarageVehicleContext context, IConfiguration iConfig)
+        IConfiguration _config;
```
Comment (IronPikachu) [inline Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: This field is unnecessary, we wanted it removed from the beginning if I recall correctly.

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs [hunk 3]
```diff
@@ -12,14 +12,15 @@ namespace Garage_2._0.Controllers.VehiclesController
...
+        public VehiclesController(GarageVehicleContext context,IConfiguration config)
         {
             _context = context;
-            _iConfig = iConfig;
+            _config = config;
+            Global.Garagecapacity = _config.GetValue<int>("GarageCapacity:Capacity");
+            Global.HourlyRate = _config.GetValue<double>("Price:HourlyRate");
```
Comment (IronPikachu) [inline Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: You can remove the _config = config on line 20 and replace _config with config on lines 21 and 22.

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs [hunk 4]
```diff
@@ -12,14 +12,15 @@ namespace Garage_2._0.Controllers.VehiclesController
...
         Vehicle[] parkingSpots;
-
-        IConfiguration _iConfig;
-        public VehiclesController(GarageVehicleContext context, IConfiguration iConfig)
+        IConfiguration _config;
+        public VehiclesController(GarageVehicleContext context,IConfiguration config)
         {
             _context = context;
-            _iConfig = iConfig;
+            _config = config;
```
Comment (IronPikachu) [inline Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]:
```suggestion

```

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs [hunk 5]
```diff
@@ -12,14 +12,15 @@ namespace Garage_2._0.Controllers.VehiclesController
...
+            Global.Garagecapacity = _config.GetValue<int>("GarageCapacity:Capacity");
```
Comment (IronPikachu) [inline Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]:
```suggestion
            Global.Garagecapacity = config.GetValue<int>("GarageCapacity:Capacity");
```

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs (hunk 3 above)
Comment (IronPikachu) [inline Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]:
```suggestion
            Global.HourlyRate = config.GetValue<double>("Price:HourlyRate");
```

**Review by IronPikachu — CHANGES_REQUESTED** [review]

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs (hunk 2 above)
Comment (RadhaArul) [reply Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: i have fixed that

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs (hunk 3 above)
Comment (RadhaArul) [reply Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: removed

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs (hunk 4 above)
Comment (RadhaArul) [reply Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: removed

File: Garage 2.0/Controllers/VehiclesController/VehiclesController.cs (hunk 1 above)
Comment (RadhaArul) [reply Garage 2.0/Controllers/VehiclesController/VehiclesController.cs]: global is static class so you cant initialize config there

**Review by IronPikachu — APPROVED** [review]

**Review by cisci86 — APPROVED** [review]

**Pull request merged**